src/prefect/orion/schemas/schedules.py

```diff
@@ -242,7 +263,32 @@ async def get_dates(
         start: datetime.datetime = None,
         end: datetime.datetime = None,
     ) -> List[pendulum.DateTime]:
-        """Retrieves dates from the schedule. Up to 10,000 candidate dates are checked
+        """Retrieves dates from the schedule. Up to 1,000 candidate dates are checked
+        following the start date.
+
+        Args:
+            n (int): The number of dates to generate
+            start (datetime.datetime, optional): The first returned date will be on or
+                after this date. Defaults to None.  If a timezone-naive datetime is
+                provided, it is assumed to be in the schedule's timezone.
+            end (datetime.datetime, optional): The maximum scheduled date to return. If
+                a timezone-naive datetime is provided, it is assumed to be in the
+                schedule's timezone.
+
+        Returns:
+            List[pendulum.DateTime]: a list of dates
+        """
+        return sorted(
+            [i async for i in self._get_dates_generator(n=n, start=start, end=end)]
+        )
+
+    async def _get_dates_generator(
```

Comment:
It looks like this doesn't need to be async anymore, probably shouldn't use an async generator unless we need `await`.